src/gateway/shard.rs

```diff
@@ -829,6 +834,33 @@ impl Shard {
     }
 }
 
+#[cfg(feature = "rustls_support")]
+fn connect(base_url: &str) -> Result<WsClient> {
+    let mut config = rustls::ClientConfig::new();
+    config.root_store.add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
+
+    let url = build_gateway_url(base_url)?;
+
+    let base_host = if let Some(h) = url.host_str() {
+        let (dot, _) = h.rmatch_indices('.').skip(1).next().unwrap_or((0, ""));
+        let (_, base) = h.split_at(dot + 1); //We do not want the leading '.'
```

Comment:
Nitpick, space after the double slashes:
```suggestion
        let (_, base) = h.split_at(dot + 1); // We do not want the leading '.'
```